solidity/test/token_stake/TestTokenStakingEscrow.js

```diff
@@ -413,6 +415,85 @@ describe('TokenStakingEscrow', () => {
       })
     })
   })
+
+  describe('withdrawRevoked', async () => {
+    const depositedAmount = 10000
+    beforeEach(async () => {
+      const data = web3.eth.abi.encodeParameters(
+        ['address', 'uint256'], [operator, grantId]
+      )
+      await token.approveAndCall(
+        escrow.address, depositedAmount, data, {from: tokenStaking}
+      )
+    })
+
+    it('can be called by grant manager', async () => {
+      await tokenGrant.revoke(grantId, {from: grantManager})
+      await escrow.withdrawRevoked(operator, {from: grantManager})
+      // ok, no reverts
+    })
+
+    it('can be called by grantee', async () => {
+      await tokenGrant.revoke(grantId, {from: grantManager})
+      await escrow.withdrawRevoked(operator, {from: grantee})
+      // ok, no reverts
+    })
+
+    it('can be called by operator', async () => {
+      await tokenGrant.revoke(grantId, {from: grantManager})
+      await escrow.withdrawRevoked(operator, {from: operator})
+      // ok, no reverts
+    })
+
+    it('can be called by third party', async () => {
+      await tokenGrant.revoke(grantId, {from: grantManager})
+      await escrow.withdrawRevoked(operator, {from: thirdParty})
+      // ok, no reverts
+    })
+
+    it('can not be called for non-revoked grant', async () => {
+      await expectRevert(
+        escrow.withdrawRevoked(operator, {from: grantManager}),
+        "Grant not revoked"
+      )
+    })
+
+    it('withdraws part of deposited amount if something has been withdrawn before', async () => {
+      await time.increaseTo(grantStart.add(time.duration.days(15)))
+      await escrow.withdraw(operator, {from: operator}) // (1000 / 30) * 15 = 5000
+      await tokenGrant.revoke(grantId, {from: grantManager})
+
+      const balanceBefore = await token.balanceOf(grantManager)
+      await escrow.withdrawRevoked(operator, {from: grantManager})
+      const balanceAfter = await token.balanceOf(grantManager)
+
+      const diff = balanceAfter.sub(balanceBefore)
+      expect(diff).to.eq.BN(5000) // 10000 - 5000 = 5000
+    })
+
+    it('withdraws entire deposited amount if nothing has been withdrawn before', async () => {
```

Comment:
Do we want to add a case where the whole amount has already been withdrawn but `withdrawRevoked` function is called again?

Reply:
:+1: I'll add similar test to `withdrawToManagedGrantee` and `withdraw`.